coremltools/converters/nnssa/frontend/graph_pass/type_inference.py

```diff
@@ -1120,6 +1120,26 @@ def _shape_transposed(self, shape):
         shape[-1], shape[-2] = shape[-2], shape[-1]
         return tuple(shape)
 
+    def visit_Einsum(self, node):
+
+        if len(node.inputs) > 2:
+            raise ValueError('No support for more than 2 inputs to {} node {} now.'.format(node.op, node.name))
+        output_shape = node.attr.get("_output_shapes")
+        if not len(output_shape) == 1:
+            raise ValueError('Expect only one output for Einsum.')
+
+        assert(len(output_shape) == 1)
```

Comment:
This `assert` seems redundant? (you already have the `if` and `raise` above.)

Reply:
good catch!

Reply:
Yes but I guess what I mean is here : 
https://github.com/apple/coremltools/blob/4704fd7c8b0643c4514e5f6fbd001d76caf4c7ef/coremltools/models/neural_network/builder.py#L1014
This layer is used to do inner product between a matrix and a weight W :) 
I didn't find a builder API to do inner product between two matrices.
Yes we can change one matrix to weight (?) but I think that would be a little bit confusing :)